src/main/java/com/icthh/xm/ms/entity/repository/search/PermittedSearchRepository.java

```diff
@@ -52,6 +55,40 @@
         return getElasticsearchTemplate().queryForPage(buildQuery(query, pageable, privilegeKey), entityClass);
     }
 
+    /**
+     * Search permitted entities with scroll
+     * @param scrollTimeInMillis The time in millisecond for scroll feature
+     * @param query the elastic query
+     * @param pageable the page info
+     * @param entityClass the search entity class
+     * @param privilegeKey the privilege key
+     * @return permitted entities
+     */
+    public <T> Page<T> search(Long scrollTimeInMillis,
+                              String query,
+                              Pageable pageable,
+                              Class<T> entityClass,
+                              String privilegeKey) {
+
+        ScrolledPage<T> scrollResult = (ScrolledPage<T>) getElasticsearchTemplate()
+            .startScroll(scrollTimeInMillis, buildQuery(query, pageable, privilegeKey), entityClass);
+
+        String scrollId = scrollResult.getScrollId();
+        List<T> resultList = new ArrayList<>();
+
+        while (scrollResult.hasContent()) {
+            resultList.addAll(scrollResult.getContent());
+            scrollId = scrollResult.getScrollId();
+
+            scrollResult = (ScrolledPage<T>) getElasticsearchTemplate()
+                .continueScroll(scrollId, scrollTimeInMillis, entityClass);
+        }
+
+        getElasticsearchTemplate().clearScroll(scrollId);
```

Comment:
what would happen if we will not call clearScroll() in case of exception?
should we use try with finally for that ?

Reply:
by documentation:

> The search contexts used by the Search Scroll API are automatically deleted when the scroll times out. But it is advised to release search contexts as soon as they are not necessary anymore using the Clear Scroll API

Reply:
than surround with 
try {
...
} finally {}